lib/dicom_net/endpoint.ex

```diff
@@ -72,8 +76,8 @@ defmodule DicomNet.Endpoint do
   end
 
   @impl true
-  def handle_cast({:register_cfind_getresponses, function}, state) do
-    new_state = %{state | getresponses: function}
+  def handle_cast({:register_handlers, handlers}, state) do
+    new_state = %{state | handlers: handlers}
```

Comment:
Is there a reason you left this message-based registration in the endpoint instead of also just making this a constructor argument? Don't really see a benefit in it anymore.